ambari-server/src/main/java/org/apache/ambari/server/topology/DownloadMpacksTask.java

```diff
@@ -72,9 +73,11 @@ public void downloadMissingMpacks(Collection<MpackInstance> mpackInstances) {
         }
       }
       catch (Exception ex) {
-        throw ex instanceof RuntimeException ?
-          (RuntimeException)ex :
-          new RuntimeException("Error occured while registering mpack: " + mpack.getStackId(), ex);
+        Throwable rootCause = ExceptionUtils.getRootCause(ex);
```

Comment:
Is it possible for `rootCause` to be `null`?

Reply:
Good catch, fixed.